bazel/canbench.bzl

```diff
@@ -63,3 +64,15 @@ def rust_canbench(name, results_file, **kwargs):
         env = env,
         args = ["--update"],
     )
+
+    if add_test:
+        native.sh_test(
+            name = name + "_test",
+            srcs = [
+                "//bazel:canbench.sh",
+            ],
+            data = data,
+            env = env,
+            tags = ["requires-network"],
```

Comment:
Without this canbench fails to download pocket-ic: https://github.com/dfinity/canbench/blob/5f8136247814cc683adb1ea3c2b503c1b8510f4a/canbench-bin/src/lib.rs#L145
I am not sure why it is not needed for the other targets... Maybe `sh_binary` has access to the local file (pocket-ic-x86_64-linux.gz, which canbench uses if available) and `sh_test` doesn't?

Reply:
@ielashi regarding [this code](https://github.com/dfinity/canbench/blob/5f8136247814cc683adb1ea3c2b503c1b8510f4a/canbench-bin/src/lib.rs#L101-L120), is there such a tight coupling between canbench and pocket-ic that you need to match on hashes? Or would a minimal version check be good enough. If the latter is possible we could provide the HEAD version of pocket-ic as a Bazel dep in the above Bazel rule.

Reply:
@basvandijk canbench checks the hashes to ensure that, for a given version of canbench, the result of a benchmark execution is deterministic. We could potentially add a flag to override that entirely and to specify a path to the Pocket IC path to use instead.

Reply:
@ielashi a flag like that would be nice. We can guarantee deterministic results by pinning pocket-ic with Bazel.

Reply:
I assume this would require a new release of canbench? Should we wait with this PR for the new canbench version or merge with network access flag for now?

Reply:
I would rather see this being fixed now. In my experience targets that require the network have a way higher chance of being / becoming flaky. Also, it happens often that we intend to fix things later but then it never happens so I think it's better to bite the bullet now. It's probably a few lines of code that need to be added in canbench to support this.

Reply:
Yes, no problem at all. I can get this done. To be clear, I'll add two flags:

1. A flag to skip the hash check
2. A flag to specify the path of pocket IC, as maybe that'll make it more ergonomic for your use-case.

Reply:
And I also agree with @basvandijk to get this done right from the get go. A new release of canbench isn't a big deal, and shouldn't be perceived as such.

Reply:
Created the PR in canbench as discussed: https://github.com/dfinity/canbench/pull/66

Reply:
FYI I didn't forget about this. There were a couple of minor features that needed to be added in there as well for stability. I'm working on getting those PRs merged and creating a new release.

Reply:
With https://github.com/dfinity/ic/pull/1468 on the way we should also think about with which pocket-ic server we want to run the canbench benchmarks with: mainnet or HEAD. Since all pocket-ic-based tests now use mainnet on PRs we should probably pick that one for benchmarks as well. Or have both.

Reply:
I think the most stable setup would be to use the version of PocketIC that canbench expects. Otherwise, this job might be a bit flaky. Any change to the replica's performance will cause the benchmarks here to fail CI, which would be an annoyance for the protocol teams. There's also the assumption that any version of Pocket IC would work with any version of canbench, which should be true in theory, but may not necessarily hold true in the future.

In contrast, if we fix the pocket IC version to what canbench expects, protocol-level changes wouldn't affect the benchmarks here until canbench/pocket IC are updated.

Reply:
@basvandijk @maciejdfinity Apologies for the delay here. canbench 0.1.6 is out. You can now run `canbench --runtime-path ...` and have that point to the path to Pocket IC. As I mentioned earlier, I recommend sticking with the version of Pocket IC that canbench expects to keep benchmarks stable and to not make tests flaky.

Reply:
Thanks for adding this! I tried using the head pocket-ic in this [commit](https://github.com/dfinity/ic/pull/1347/commits/ad02a8278d8e1c2c4566574116787f0ff9c0ec08), but the error I get suggests pocket-ic 4.0.0 is still used:
```
CANBENCH_BIN: external/crate_index__canbench-0.1.6/canbench__bin
POCKET_IC_BIN: rs/pocket_ic_server/pocket-ic-server
pocket ic version:
pocket-ic-server 6.0.0
thread 'main' panicked at external/crate_index__pocket-ic-4.0.0/src/lib.rs:1315:13:
Failed to start PocketIC service in time
```
Am I using the params correctly? Should I use another version?